spec/requests/embeddings/embeddings_controller_spec.rb

```diff
@@ -60,6 +60,19 @@ def create_api_key(user)
       expect(response.parsed_body["topics"].map { |t| t["id"] }).to contain_exactly(topic.id)
     end
 
+    context "when rate limiting is enabled" do
+      before { RateLimiter.enable }
+
+      it "will not rate limit API for hyde search" do
+        10.times do |i|
+          query = "test #{SecureRandom.hex}"
+          stub_embedding(query)
+          get "/discourse-ai/embeddings/semantic-search.json?q=#{query}&hyde=false"
+          expect(response.status).to eq(200)
+        end
```

Comment:
Maybe we can stub the `MAX_HYDE_SEARCHES_PER_MINUTE ` constant so that we don't have to trigger 10 requests here?

Reply:
ill time the test and see if it is worth a follow up